misc/xml-to-record-converter/src/main/java/io/ballerina/xmltorecordconverter/XMLToRecordConverter.java

```diff
@@ -0,0 +1,372 @@
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.compiler.syntax.tree.Token;
+import io.ballerina.compiler.syntax.tree.TypeDefinitionNode;
+import io.ballerina.compiler.syntax.tree.TypeDescriptorNode;
+import io.ballerina.xmltorecordconverter.diagnostic.DiagnosticMessage;
+import io.ballerina.xmltorecordconverter.diagnostic.DiagnosticUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.ballerinalang.formatter.core.ForceFormattingOptions;
+import org.ballerinalang.formatter.core.Formatter;
+import org.ballerinalang.formatter.core.FormatterException;
+import org.ballerinalang.formatter.core.FormattingOptions;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.xml.sax.SAXException;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import static io.ballerina.xmltorecordconverter.util.ConverterUtils.escapeIdentifier;
+import static io.ballerina.xmltorecordconverter.util.ConverterUtils.extractUnionTypeDescNode;
+import static io.ballerina.xmltorecordconverter.util.ConverterUtils.getPrimitiveTypeName;
+import static io.ballerina.xmltorecordconverter.util.ConverterUtils.sortTypeDescriptorNodes;
+
+/**
+ * APIs for conversion from XML to Ballerina records.
+ *
+ * @since 2201.7.2
+ */
+public class XMLToRecordConverter {
+
+    private XMLToRecordConverter() {}
+
+    public static XMLToRecordResponse convert(String xmlValue, boolean preserveNamespaces, boolean isRecordTypeDesc,
+                                              boolean isClosed, boolean forceFormatRecordFields) {
+        Map<String, NonTerminalNode> recordToTypeDescNodes = new LinkedHashMap<>();
+        List<DiagnosticMessage> diagnosticMessages = new ArrayList<>();
+        XMLToRecordResponse response = new XMLToRecordResponse();
+
+        try {
+            // Convert the XML string to an InputStream
+            ByteArrayInputStream inputStream = new ByteArrayInputStream(xmlValue.getBytes(StandardCharsets.UTF_8));
+
+            DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder docBuilder = dbFactory.newDocumentBuilder();
+            Document doc = docBuilder.parse(inputStream);
+
+            Element rootElement = doc.getDocumentElement();
+            generateRecords(rootElement, isClosed, recordToTypeDescNodes, diagnosticMessages);
+        } catch (ParserConfigurationException parserConfigurationException) {
+
+        } catch (SAXException saxException) {
+
+        } catch (IOException | IllegalArgumentException e) {
+
+        }
```

Comment:
Shall we handle these errors and return diagnostics, similar to what we are doing during the JSON to record conversion?